v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+            case TlgInINT8:
+                parsed_events[current_parsed_event_count].properties[property_index].int8_t_value = *(int8_t*)current_user_data_field;
+                current_user_data_field += sizeof(int8_t);
+                break;
+            case TlgInUINT16:
+                parsed_events[current_parsed_event_count].properties[property_index].uint16_t_value = *(uint16_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint16_t);
+                break;
+            case TlgInINT16:
+                parsed_events[current_parsed_event_count].properties[property_index].int16_t_value = *(int16_t*)current_user_data_field;
+                current_user_data_field += sizeof(int16_t);
+                break;
+            case TlgInUINT32:
+                parsed_events[current_parsed_event_count].properties[property_index].uint32_t_value = *(uint32_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint32_t);
+                break;
+            case TlgInINT32:
+                parsed_events[current_parsed_event_count].properties[property_index].int32_t_value = *(int32_t*)current_user_data_field;
+                current_user_data_field += sizeof(int32_t);
+                break;
+            case TlgInUINT64:
+                parsed_events[current_parsed_event_count].properties[property_index].uint64_t_value = *(uint64_t*)current_user_data_field;
+                current_user_data_field += sizeof(uint64_t);
+                break;
+            case TlgInINT64:
+                parsed_events[current_parsed_event_count].properties[property_index].int64_t_value = *(int64_t*)current_user_data_field;
+                current_user_data_field += sizeof(int64_t);
+                break;
+            }
+
+            property_index++;
+        }
+
+        parsed_events[current_parsed_event_count].property_count = property_index;
+
+        (void)InterlockedIncrement(&parsed_event_count);
+        WakeByAddressAll((void*)&parsed_event_count);
+    }
+}
+
+static char test_path[MAX_PATH];
+
+static void generate_trace_session(void)
+{
+    UUID uuid;
+    POOR_MANS_ASSERT(UuidCreate(&uuid) == RPC_S_OK);
+    int snprintf_result = snprintf(trace_session_name, sizeof(trace_session_name), "%s-%08X-%04X-%04X-%02X%02X-%02X%02X%02X%02X%02X%02X",
+        __FUNCTION__, uuid.Data1, uuid.Data2, uuid.Data3, uuid.Data4[0], uuid.Data4[1], uuid.Data4[2], uuid.Data4[3], uuid.Data4[4], uuid.Data4[5], uuid.Data4[6], uuid.Data4[7]);
+    POOR_MANS_ASSERT((snprintf_result > 0) && (snprintf_result < sizeof(trace_session_name)));
+}
+
+static TRACEHANDLE start_trace(uint8_t trace_level)
+{
+    // setup the etw consumer
+    TRACEHANDLE trace_session_handle;
+
+    (void)printf("Starting trace session %s\r\n", trace_session_name);
+
+    EVENT_TRACE_PROPERTY_DATA start_event_trace_property_data = { 0 };
+
+    start_event_trace_property_data.props.Wnode.BufferSize = sizeof(start_event_trace_property_data);
```

Comment:
>Wnode [](http://example.com/codeflow?start=42&length=5)

This WNODE structure has a lot of other information in it ([https://learn.microsoft.com/en-us/windows/win32/etw/wnode-header](https://learn.microsoft.com/en-us/windows/win32/etw/wnode-header)) but I will assume these are the only ones needed for the purpose here.

Reply:
I assume the same.  started from a sample that only had this.